packages/core/admin/admin/src/content-manager/pages/ListView/components/FieldPicker/index.js

```diff
@@ -1,15 +1,32 @@
 import React from 'react';
 
-import { Select, Option, Box } from '@strapi/design-system';
+import { Box, Flex, Checkbox, TextButton, Typography } from '@strapi/design-system';
 import { useTracking } from '@strapi/helper-plugin';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 import { useDispatch, useSelector } from 'react-redux';
+import styled, { css } from 'styled-components';
 
-import { getTrad, checkIfAttributeIsDisplayable } from '../../../../utils';
-import { onChangeListHeaders } from '../../actions';
+import { checkIfAttributeIsDisplayable } from '../../../../utils';
+import { onChangeListHeaders, onResetListHeaders } from '../../actions';
 import { selectDisplayedHeaders } from '../../selectors';
 
+const activeCheckboxWrapperStyles = css`
+  background: ${(props) => props.theme.colors.primary100};
+  svg {
+    opacity: 1;
+  }
+  border-radius: ${({ theme }) => theme.borderRadius};
+`;
+
+const CheckboxWrapper = styled(Box)`
+  /* Show active style both on hover and when the action is selected */
+  ${(props) => props.isActive && activeCheckboxWrapperStyles}
+  &:hover {
+    ${activeCheckboxWrapperStyles}
+  }
+`;
+
```

Comment:
I'd prefer if you could inline the `svg` overwrite in `CheckboxWrapper`. All other attributes can become props (because `CheckboxWrapper` inherits from `Box`), which will simplify this a lot.

Reply:
Thanks for your suggestion but I don't think it works, 
I tried to move the background color and borderRadius definition as props but it doesn't work 
```
          <CheckboxWrapper
            padding={2}
            width="100%"
            key={header.name}
            isActive={isSelected(header.name)}
            backgroundColor="primary100"
            borderRadius="borderRadius"
          >
```
And also the style is applied if the checkbox is active or we are on hover, I can apply the condition active in the props but not the hover condition
I know you don't like the definition of `activeCheckboxWrapperStyles` but to be honest I think it is good to avoid duplications and it is largely used this condition inside the code

Reply:
```
<CheckboxWrapper backgroundColor={isActive ? "primary100" : "something else"} />
```

does not work in your case?

I agree about the `:hover` and nested `svg` role - they need to be overwritten but background-color and border-radius shouldn't I think.

Reply:
I tried your solution but it doesn't work in my case unfortunately

Reply:
Ok, I'll have a look in the afternoon then and see if I can make it work 👍🏼

Reply:
I've done some refactoring in 8be603a2dd3e5548e2661e448cd79b030e31bfc1.

Reply:
Thanks @gu-stav I saw your fixes and why they work, I missed the correct name for the background prop. Thanks also for the refactoring in the FieldPicker component